### ISSUE TITLE
Cluster Autoscaler can't remove not replicated pod

### ISSUE THREAD

**Issue by yancl** [riseml/riseml#11]
Hi,
  when using riseml we meet such problem and it is triggered by the following steps:
  * run a task(A) which needs 4 gpu
  * cluster autoscaler start a new p3.8xlarge node
  * run a task(B) which needs no gpu but some cpu
  * task B is scheduled to the same node as A
  * task A finished
  * task B needs many hours
  * cluster autoscaler can't replcace the **big** node with a **small** node because task B is using it and task B **doesn't** have **replica controller** with it(I think the task is managed by riseml system)

  So my question is:
  * Will the task re-run automatically just like hadoop jobs if the task is killed before finished
  * Will riseml add some annotations to pod like [this](https://github.com/kubernetes/autoscaler/pull/418) to work around the problem if question one is YES

thanks.

**Comment by yancl** [comment]
ping @elmarhaussmann :)

**Comment by elmarhaussmann** [comment]
@yancl : yes, this seems to be the problem: the node will not be drained by Kubernetes and scaled down if the Pod running on it is not replicated. We will add this to our backlog as adding this on our end seems to be the only solution. We will update here once we solved it.

**Comment by yancl** [comment]
thanks :-)
